app/controllers/likes_controller.rb

```diff
@@ -57,13 +50,24 @@ def destroy
   end
 
   private
-    # Use callbacks to share common setup or constraints between actions.
-    def set_like
-      @like = Like.find(params[:id])
-    end
 
-    # Only allow a list of trusted parameters through.
-    def like_params
-      params.require(:like).permit(:fan_id, :photo_id)
-    end
+  def set_photo
+    @photo = Photo.find(params[:like][:photo_id])
+  end
+
+  def set_like
+    @like = Like.find(params[:id])
+  end
+
+  def authorize_resource
+    authorize @like || Like
+  end
+
+  def authorize_create_action
```

Comment:
i don't think you need this method abstraction.

Reply:
removed